src/Components/Browser/src/RendererRegistryEventDispatcher.cs

```diff
@@ -30,34 +37,61 @@ private static UIEventArgs ParseEventArgsJson(string eventArgsType, string event
             switch (eventArgsType)
             {
                 case "change":
-                    return Json.Deserialize<UIChangeEventArgs>(eventArgsJson);
+                    return DeserializeUIEventChangeArgs(eventArgsJson);
                 case "clipboard":
-                    return Json.Deserialize<UIClipboardEventArgs>(eventArgsJson);
+                    return Deserialize<UIClipboardEventArgs>(eventArgsJson);
                 case "drag":
-                    return Json.Deserialize<UIDragEventArgs>(eventArgsJson);
+                    return Deserialize<UIDragEventArgs>(eventArgsJson);
                 case "error":
-                    return Json.Deserialize<UIErrorEventArgs>(eventArgsJson);
+                    return Deserialize<UIErrorEventArgs>(eventArgsJson);
                 case "focus":
-                    return Json.Deserialize<UIFocusEventArgs>(eventArgsJson);
+                    return Deserialize<UIFocusEventArgs>(eventArgsJson);
                 case "keyboard":
-                    return Json.Deserialize<UIKeyboardEventArgs>(eventArgsJson);
+                    return Deserialize<UIKeyboardEventArgs>(eventArgsJson);
                 case "mouse":
-                    return Json.Deserialize<UIMouseEventArgs>(eventArgsJson);
+                    return Deserialize<UIMouseEventArgs>(eventArgsJson);
                 case "pointer":
-                    return Json.Deserialize<UIPointerEventArgs>(eventArgsJson);
+                    return Deserialize<UIPointerEventArgs>(eventArgsJson);
                 case "progress":
-                    return Json.Deserialize<UIProgressEventArgs>(eventArgsJson);
+                    return Deserialize<UIProgressEventArgs>(eventArgsJson);
                 case "touch":
-                    return Json.Deserialize<UITouchEventArgs>(eventArgsJson);
+                    return Deserialize<UITouchEventArgs>(eventArgsJson);
                 case "unknown":
-                    return Json.Deserialize<UIEventArgs>(eventArgsJson);
+                    return Deserialize<UIEventArgs>(eventArgsJson);
                 case "wheel":
-                    return Json.Deserialize<UIWheelEventArgs>(eventArgsJson);
+                    return Deserialize<UIWheelEventArgs>(eventArgsJson);
                 default:
                      throw new ArgumentException($"Unsupported value '{eventArgsType}'.", nameof(eventArgsType));
             }
         }
 
+        private static T Deserialize<T>(string eventArgsJson)
+        {
+            return JsonSerializer.Parse<T>(eventArgsJson, SerializerOptions);
+        }
+
+        private static UIChangeEventArgs DeserializeUIEventChangeArgs(string eventArgsJson)
+        {
+            var changeArgs = Deserialize<UIChangeEventArgs>(eventArgsJson);
+            var jsonElement = (JsonElement)changeArgs.Value;
+            switch (jsonElement.Type)
+            {
+                case JsonValueType.Null:
```

Comment:
should we allow undefined and treat the same as null? or not going to show up here?

Reply:
@ahsonkhan \ @JeremyKuhne when would JsonElement return `Undefined`?

Reply:
When there was no value in the Json.

Reply:
For example, if you had a default instance of `JsonElement` or if the underlying document was disposed.

I would consider that to be an exceptional case, so throwing is probably best.

cc @bartonjs

Reply:
Leaving this as-is for now. We can wait for a user report since none of the tests currently run in to an undefined value.

Reply:
Unless someone explicitly wrote `JsonElement element = default;` the only practical way they'd see a JsonElement whose Type was Undefined is reading the out-value from TryGetProperty when it returned false (because the API internally said `value = default;`).